core/templates/components/rubrics-editor/rubrics-editor.component.ts

```diff
@@ -91,8 +91,30 @@ export class RubricsEditorComponent {
     this.explanationEditorIsOpen[difficulty][index] = true;
   }
 
-  isExplanationValid(difficulty: string, index: number): boolean {
-    return Boolean(this.editableExplanations[difficulty][index]);
+  isExplanationLengthValid(difficulty: string, idx: number): boolean {
+    return this.editableExplanations[difficulty][idx].length <= 300;
+  }
+
+  isMediumLevelExplanationValid(): boolean {
+    if (this.rubrics[1] === undefined) {
+      return false;
+    }
+    return (this.rubrics[1]._explanations.length >= 1);
+  }
+
+  isTotalExplanationsLengthReached(): boolean {
+    if (this.rubrics[0] === undefined || this.rubrics[1] === undefined ||
+      this.rubrics[2] === undefined) {
+      return false;
+    } else {
+      let totalEasyExplanations = this.rubrics[0]._explanations.length;
+      let totalMediumExplanations = this.rubrics[1]._explanations.length;
+      let totalHardExplanations = this.rubrics[2]._explanations.length;
+
+      return (
+        totalEasyExplanations + totalMediumExplanations +
+        totalHardExplanations >= 10);
+    }
```

Comment:
ditto: avoid "magic numbers", avoid accessing private variables and prefer getters, avoid hardcoding indices.

Reply:
Done, Please do let me know if this approach is okay.
Thanks